pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1489,8 +1486,9 @@ func reportGatewayStatus(
 
 	// Setup initial conditions to the success state. If we encounter errors, we will update this.
 	// We have two status
-	// Accepted: did we get *some* data to the data plane
-	// Programmed: did we get *all* the data to the data plane
+	// Accepted: is the configuration valid. We only have errors in listeners, and the status is not supposed to
```

Comment:
I guess "Accepted" really means just that - the gateway controller associated with the specified gatewayClass has seen it. It's not really validating anything at this point since any other validation errors (preventing acceptance) would have prevented it from being applied in the first place. The only thing that could have been wrong is an incorrect gatewayClass, which the controller is confirming valid by virtue of having found it. Does that sound right?

Reply:
Basically yes, its Gateway level static validation. So generally for us that just means "we saw it" since we don't have anything to validate (at least right now)

Reply:
Would it be possible to reuse any of existing status of Kubernetes? Or is this status part of kubernetes gateway API? I feel either `accepted` or `programmed` requires reading docs to understand

Reply:
This is part of the gateway API. They define them, we just implement.

Reply:
I tried to change it to ready but got told a resounding "no"

Reply:
ok looks like these are gateway conditions, not status and part of existing conditions from gateway API.